File: app/src/main/java/com/sam/shamsaha/resources/percountry/adapter/ResourcesDetailsAdapter.java
package com.sam.shamsaha.resources.percountry.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.sam.shamsaha.R;

public class ResourcesDetailsAdapter extends RecyclerView.Adapter<ResourcesDetailsViewHolder> {


    private Context context;

    public ResourcesDetailsAdapter(Context context) {
        this.context = context;
    }

    @NonNull
    @Override
    public ResourcesDetailsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.resources_view_holder, parent, false);
        return  new ResourcesDetailsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ResourcesDetailsViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 15;
    }
}
class ResourcesDetailsViewHolder extends RecyclerView.ViewHolder {

    public ResourcesDetailsViewHolder(@NonNull View itemView) {
        super(itemView);
    }
}

File: app/src/main/java/com/sam/shamsaha/volunteer/resources/ResourcesDetailsAdapter.kt
package com.sam.shamsaha.volunteer.resources

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.NonNull
import androidx.recyclerview.widget.RecyclerView
import com.sam.shamsaha.databinding.SingleItemBinding

class ResourcesDetailsAdapter(val context: Context, val resultDetailsList: List<ValDetailsModel>,):
    RecyclerView.Adapter
    <ResourcesDetailsAdapter.ResourceDetailsViewHolder>() {


    inner class ResourceDetailsViewHolder(val binding: SingleItemBinding) : RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ResourceDetailsViewHolder {
        val binding = SingleItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ResourceDetailsViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ResourceDetailsViewHolder, position: Int) {

    }

    override fun getItemCount(): Int {
        return 10;
    }

}